zokrates_core/src/embed.rs

```diff
@@ -281,6 +321,102 @@ pub fn sha256_round<T: Field>() -> FlatFunction<T> {
     }
 }
 
+#[cfg(feature = "ark")]
+pub fn verify<T: Field>(n: usize) -> FlatFunction<T> {
+    let (out_index, input_indices, proof_indices, vk_indices, constraints, variable_count) =
+        generate_verify_constraints(n);
+
+    let cs_indices = 0..variable_count;
+    let input_indices = input_indices.into_iter();
+    let proof_indices = proof_indices.into_iter();
+    let vk_indices = vk_indices.into_iter();
+
+    // indices of the arguments to the function
+    let input_argument_indices = input_indices.clone().map(|i| i + variable_count);
+
+    let proof_argument_indices = proof_indices.clone().map(|i| i + variable_count);
+
+    let vk_argument_indices = vk_indices.clone().map(|i| i + variable_count);
+
+    let input_arguments = input_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let proof_arguments = proof_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let vk_arguments = vk_argument_indices
+        .clone()
+        .map(|i| FlatParameter::private(FlatVariable::new(i)));
+
+    let arguments = input_arguments
+        .chain(proof_arguments)
+        .chain(vk_arguments)
+        .collect();
+
+    let one_binding_statement = FlatStatement::Condition(
+        FlatExpression::Identifier(FlatVariable::new(0)),
+        FlatExpression::Number(T::from(1)),
+    );
+
+    let input_binding_statements: Vec<_> = input_indices
+        .chain(proof_indices)
+        .chain(vk_indices)
+        .zip(
+            input_argument_indices
+                .clone()
+                .chain(proof_argument_indices.clone())
+                .chain(vk_argument_indices.clone()),
+        )
+        .map(|(cs_index, argument_index)| {
+            FlatStatement::Condition(
+                FlatVariable::new(cs_index).into(),
+                FlatVariable::new(argument_index).into(),
+            )
+        })
+        .collect();
+
+    let constraint_statements: Vec<FlatStatement<T>> = constraints
+        .into_iter()
+        .map(|c| {
+            let c: Constraint<T> = from_ark::<T, Bls12_377>(c);
```

Comment:
Since we're using a specific curve here, this function should check the id at the top